src/nvim/main.c

```diff
@@ -1444,6 +1444,20 @@ static void command_line_scan(mparm_T *parmp)
       ga_grow(&global_alist.al_ga, 1);
       char *p = xstrdup(argv[0]);
 
+      // On Windows expand "~\" or "~/" prefix in file names to profile directory.
+#ifdef MSWIN
+      if (*p == '~' && (p[1] == '\\' || p[1] == '/')) {
+        const char *profile_dir = os_getenv("USERPROFILE");
+        if (profile_dir) {
```

Comment:
On second thought, it may be better to use `vim_getenv("HOME")` here. That'll return the `homedir` variable in os/env.c, and there is no need to check for `NULL`.